From: Wolfgang Bauer <wbauer@tmo.at>
Date: Fri, 22 Apr 2016 17:23:35 +0000
Subject: Always write the theme name to plasmarc, also if the default theme is chosen
X-Git-Url: http://quickgit.kde.org/?p=plasma-framework.git&a=commitdiff&h=d884119f5a4a618eb30baaf5cd0ef058269b27f3
---
Always write the theme name to plasmarc, also if the default theme is chosen

Currently, the "name" key is deleted from plasmarc when you select the
default theme (breeze) in the desktop theme KCM.
This leads to unexpected behaviour: If you set e.g. oxygen as look&feel,
and then set breeze as desktop theme, Plasma actually uses Air (the
default desktop theme in the oxygen l&f) on next login instead of
breeze. In other words, it is impossible to configure breeze as desktop
theme when using a different look&feel as breeze.

This fixes the problem by removing the check whether the default theme
is selected and always writes the "name" config key to plasmarc.

REVIEW: 127698
---


--- a/src/plasma/private/theme_p.cpp
+++ b/src/plasma/private/theme_p.cpp
@@ -846,11 +846,7 @@
     if (realTheme && isDefault && writeSettings) {
         // we're the default theme, let's save our state
         KConfigGroup &cg = config();
-        if (ThemePrivate::defaultTheme == themeName) {
-            cg.deleteEntry("name");
-        } else {
-            cg.writeEntry("name", themeName);
-        }
+        cg.writeEntry("name", themeName);
         cg.sync();
     }
 

